examples/chip-tool/commands/icd/ICDCommand.cpp

```diff
@@ -64,13 +65,73 @@ CHIP_ERROR ICDListCommand::RunCommand()
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ICDWaitForDeviceCommand::RunCommand()
+{
+    if (!IsPeerLIT())
+    {
+        ChipLogError(chipTool, "The device is not a registered LIT-ICD device.");
+        return CHIP_ERROR_NOT_FOUND;
+    }
+    mInterestedNode = ScopedNodeId(GetDestinationId(), CurrentCommissioner().GetFabricIndex());
+    ChipLogError(chipTool, "Please trigger the device active mode.");
+    return CHIP_NO_ERROR;
+}
+
+void ICDWaitForDeviceCommand::OnCheckInComplete(const chip::app::ICDClientInfo & clientInfo)
+{
+    DefaultCheckInDelegate::OnCheckInComplete(clientInfo);
+
+    if (clientInfo.peer_node != mInterestedNode)
+    {
+        ChipLogDetail(chipTool, "The node " ChipLogFormatScopedNodeId " is not the one we are interested in.",
+                      ChipLogValueScopedNodeId(clientInfo.peer_node));
+        return;
+    }
+
+    ChipLogDetail(chipTool, "Received check-in message from the node, send stay active request to the device.");
+    mInterestedNode = ScopedNodeId();
+
+    // Intentionally call RunCommand, since it includes all necessary steps for SendCommand.
+    CHIP_ERROR err = ClusterCommand::RunCommand();
+    if (err != CHIP_NO_ERROR)
+    {
+        SetCommandExitStatus(err);
+        return;
+    }
+}
+
+CHIP_ERROR ICDWaitForDeviceCommand::SendCommand(chip::DeviceProxy * device,
+                                                std::vector<chip::EndpointId> /* not used, always send to endpoint 0 */)
+{
+    chip::app::Clusters::IcdManagement::Commands::StayActiveRequest::Type request;
```

Comment:
```suggestion
    Clusters::IcdManagement::Commands::StayActiveRequest::Type request;
```
since we have that `using` bit anyway, yes?